### DISCUSSION TITLE
Should the names like "Rust" be left as is or should they be translated into language but with the same pronounciation?

### DISCUSSION THREAD

**Discussion by abhik-99** [google/comprehensive-rust#1888] [General]
So I am trying to contribute to Bengali localization and on a brief discussion with @mirwasi and @mgeisler this question arose. This rings especially true when the language does not have Latin characters ( like Chinese, Japanese and Bengali). Would love a discussion on this so that we can reach a consensus and avoid future translation confusion. Example of such words would be "Rust", "Chromium", "Google" and so on.

**Comment by mirwasi** [comment]
Thanks for opening this discussion @abhik-99! I feel we need to come up with a framework (taking everyone's opinions into account).

My 2¢ (Edit: wrt Bengali): 

When it comes to names, I'd follow how the existing programming literature is written in that language. Personally, I'm a native Bengali speaker from Bangladesh and almost all of the programming literature there use the transliteration of the words in Bengali (for example: পাইথন (Python) or জাভা (Java). You can see a selection of the books available in [Bangladesh](https://www.rokomari.com/book/category/558/computer-programming?xyz=&priceRange=0to114900&languageIds=1&discountRange=0to30).

Bengali is the second largest language in India (in terms of first speakers), with a significant number of speakers. I haven't read any Bengali programming book from India myself. I was looking at Amazon India for programming related books in Bengali, but couldn't find any. @abhik-99, it would be great if you could share how programming literature in written in Bengali in India.

If a word the name of a library, or a keyword from a programming language, library, I'd try to use the term _transliterated_ in Bengali, while keeping the original term in brackets (I'll admit, I haven't been able to follow this all the time in code reviews) 

Here is a screenshot of the TOC of a very popular Python book: https://www.rokomari.com/book/224822 (No copyright infringement intended)

<img width="656" alt="pythonbook" src="https://github.com/google/comprehensive-rust/assets/1697945/a75c6da8-07a1-4c8e-9fd0-2e2e26f9db6d">

Slightly off-topic (on translation): the lack of having a standard list of programming terms in Bengali is huge problem. Dictionary meanings are often misleading and fail to convey the true meaning. Some people in my country tried creating a list of Bengali terms, but it never got the level of adoption they hoped for.

cc: @raselmandol @sudiptachatterjee

**Comment by mingyc** [comment]
> This rings especially true when the language does not have Latin characters ( like Chinese, Japanese and Bengali). 

Please note that different language/region users have different conventions. As of this topic, it does not directly correlate to whether the language use Latin characters or not. For example, it's very common in zh-TW (Traditional Chinese used in Taiwan) to use untranslated English terms, especially in the field of computer science; while in zh-CN, it's entirely different. In Japanese, terms are usually translated, but sometimes English terms/abbreviations are still used.

> Thanks for opening this discussion @abhik-99! I feel we need to come up with a framework (taking everyone's opinions into account).

There should not be such framework. Translation should also be localized.

**Reply by mirwasi** [reply]
Thanks @mingyc for your reply.

Sorry, should've been a bit clearer in my response: what I wrote was with respect to Bengali language. I agree with your general viewpoint.

**Comment by mgeisler** [comment]
> Would love a discussion on this so that we can reach a consensus and avoid future translation confusion. Example of such words would be "Rust", "Chromium", "Google" and so on.

I'm not an expert on what the conventions is around the world — and I agree with @mingyc that it's likely different from language to language.

So the only advice I can give is for each translation to come up with guidelines on this and _document them_. The documentation could be in the relevant GitHub issue, it could be in a comment in the PO file itself.

The full set of five courses is about 350 pages if you print them... that's a substantial amount of text and consistency is important for a quality translation. Documentation is thus important as well to make sure the translators are roughly on the same page :smile:
